.github/workflows/build-images.yaml

```diff
@@ -344,111 +348,105 @@ jobs:
+          PROD_AWS: "${{ vars.NEON_PROD_AWS_ACCOUNT_ID }}"
+          AWS_REGION: "${{ vars.AWS_ECR_REGION }}"
+
+  push-to-dev:
+    if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
+    needs: [ merge-images, generate-image-maps ]
 
-            image_from=${image}:${TAG}
-            image_ecr_dev="${ECR_DEV}/${image_without_prefix}:${TAG}"
-            image_ecr_prod="${ECR_PROD}/${image_without_prefix}:${TAG}"
+    permissions:
+      id-token: write  # Required for aws/azure login
+      packages: write  # required for pushing to GHCR
 
-            echo "Copy ${image_from} to ${image_ecr_dev} (dev), and ${image_ecr_prod} (prod)"
+    uses: ./.github/workflows/_push-to-container-registry.yml
+    secrets: inherit
+    with:
+      image-map: '${{ needs.generate-image-maps.outputs.dev }}'
+      aws-region: "${{ vars.AWS_ECR_REGION }}"
+      aws-account-id: "${{ vars.NEON_DEV_AWS_ACCOUNT_ID }}"
+      aws-role-to-assume: "gha-oidc-ecr-rw"
+      azure-client-id: "${{ vars.AZURE_DEV_CLIENT_ID }}"
+      azure-subscription-id: "${{ vars.AZURE_DEV_SUBSCRIPTION_ID }}"
+      azure-tenant-id: "${{ vars.AZURE_TENANT_ID }}"
+      acr-registry-name: "${{ vars.AZURE_DEV_REGISTRY_NAME }}"
+
+  push-to-prod:
+    if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
+    needs: [ merge-images, generate-image-maps ]
 
-            docker buildx imagetools create -t ${image_ecr_dev} \
-                                            -t ${image_ecr_prod} \
-                                               ${image_from}
-          done
+    permissions:
+      id-token: write  # Required for aws/azure login
+      packages: write  # required for pushing to GHCR
 
-  push-to-acr-dev:
-    if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
-    needs: [ merge-images ]
-    uses: ./.github/workflows/_push-to-acr.yaml
-    with:
-      client_id: ${{ vars.AZURE_DEV_CLIENT_ID }}
-      image_tag: ${{ inputs.tag }}
-      images: neonvm-controller neonvm-vxlan-controller neonvm-runner autoscale-scheduler autoscaler-agent cluster-autoscaler-neonvm
-      registry_name: ${{ vars.AZURE_DEV_REGISTRY_NAME }}
-      subscription_id: ${{ vars.AZURE_DEV_SUBSCRIPTION_ID }}
-      tenant_id: ${{ vars.AZURE_TENANT_ID }}
+    uses: ./.github/workflows/_push-to-container-registry.yml
     secrets: inherit
-
-  push-to-acr-prod:
-    if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
-    needs: [ merge-images ]
-    uses: ./.github/workflows/_push-to-acr.yaml
     with:
-      client_id: ${{ vars.AZURE_PROD_CLIENT_ID }}
-      images: neonvm-controller neonvm-vxlan-controller neonvm-runner autoscale-scheduler autoscaler-agent cluster-autoscaler-neonvm
-      image_tag: ${{ inputs.tag }}
-      registry_name: ${{ vars.AZURE_PROD_REGISTRY_NAME }}
-      tenant_id: ${{ vars.AZURE_TENANT_ID }}
-      subscription_id: ${{ vars.AZURE_PROD_SUBSCRIPTION_ID }}
-    secrets: inherit
+      image-map: '${{ needs.generate-image-maps.outputs.prod }}'
```

Comment:
nit: all args below use double quotes, while here single quotes, could they be aligned?
from what I see we don't need quotes here at all, but did I miss something why we need them here?

Reply:
I suspect that might not work with GitHub's variable interpolation, but I'm not 100% sure, so let's check that

Reply:
Double-quotes seems to be working: https://github.com/neondatabase/autoscaling/actions/runs/14497213379/job/40668240301?pr=1283